src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_locks.py

```diff
@@ -83,14 +107,155 @@ def _lock_operation_with_resource(self, lock_type, resource_group):
+        vnet_lock_name = self.create_random_name('cli-test-lock', 50)
+        subnet_lock_name = self.create_random_name('cli-test-lock', 20)
+
+        vnet = self.cmd('az network vnet create -n {} -g {}'.format(vnet_name, resource_group)).get_output_in_json()
+        subnetaddress = vnet.get('newVNet').get('addressSpace').get('addressPrefixes')[0]
+        self.cmd('az network vnet subnet create -n {} --address-prefix {} --vnet-name {} -g {}'
+                 .format(subnet_name, subnetaddress, vnet_name, resource_group))
+
+        locks = []
+        locks.append(self.cmd('az lock create -n {} -g {} --lock-type CanNotDelete'
+                              .format(group_lock_name, resource_group)).get_output_in_json())
+        locks.append(self.cmd('az lock create -n {} -g {} --resource-type Microsoft.Network/virtualNetworks'
+                              ' --resource-name {} --lock-type CanNotDelete'
+                              .format(vnet_lock_name, resource_group, vnet_name)).get_output_in_json())
+        locks.append(self.cmd('az lock create -n {} -g {} --resource-name {} --resource-type subnets '
+                              '--namespace Microsoft.Network --parent virtualNetworks/{} --lock-type CanNotDelete'
+                              .format(subnet_lock_name, resource_group, subnet_name, vnet_name)).get_output_in_json())
+        self._sleep_for_lock_operation()
+
+        space_delimited_ids = ' '.join([lock.get('id', None) for lock in locks])
+
+        my_locks = self.cmd('az lock show --ids {} --query [].name'.format(space_delimited_ids)).get_output_in_json()
+        self.assertTrue(len(my_locks) == 3)
+        for lock in my_locks:
+            self.assertIn(lock, [group_lock_name, vnet_lock_name, subnet_lock_name])
+
+        my_locks = self.cmd('az lock update --ids {} --notes somenotes --lock-type ReadOnly'
+                            .format(space_delimited_ids)).get_output_in_json()
+        self.assertTrue(len(my_locks) == 3)
+        for lock in my_locks:
+            self.assertEqual(lock.get('notes', None), 'somenotes')
+            self.assertEqual(lock.get('level', None), 'ReadOnly')
+
+        self.cmd('az lock delete --ids {}'.format(space_delimited_ids))
+        self._sleep_for_lock_operation()
+
+        my_locks = self.cmd("az lock list -g {} -ojson".format(resource_group)).get_output_in_json()
+        self.assertFalse(my_locks)
+
     def _sleep_for_lock_operation(self):
         if self.is_live:
             sleep(5)
 
 
+class ParseIdTests(unittest.TestCase):
+    def test_parsing_lock_ids(self):
+        tests = [
+            {
+                'input': "/subscriptions/subId/providers/"
+                         "Microsoft.Authorization/locks/sublock",
+                'expected': {
+                    'resource_group_name': None,
+                    'resource_provider_namespace': None,
+                    'parent_resource_path': None,
+                    'resource_type': None,
```

Comment:
I would expect namespace to resolve to Microsoft.Authorization and resource type to resolve to locks here.

Reply:
so in my regex, I only have values for namespace, parent, resource_type if the lock is on a resource level.
since, I am directly matching for lock ids, I considered it pointless to get at values that would be present in every lock id